src/file.rs

```diff
@@ -375,21 +375,26 @@ impl PathFile {
     ///
     /// let contents = "This is some contents";
     /// file.write_str(contents);
-    /// file.symlink(example_sym)?;
-    /// let file_sym = PathFile::new(example_sym)?;
+    /// let file_sym = file.symlink(example_sym)?;
     ///
     /// // They are canonicalized to the same file.
-    /// assert_eq!(file, file_sym);
+    /// assert_ne!(file, file_sym);
+    ///
+    ///
+    /// // But they can be canonicalized to the same file.
+    /// let file_can = file_sym.canonicalize()?;
+    /// assert_eq!(file, file_can);
     /// # Ok(()) } fn main() { try_main().unwrap() }
     /// ```
-    pub fn symlink<P: AsRef<Path>>(&self, dst: P) -> Result<()> {
+    pub fn symlink<P: AsRef<Path>>(&self, dst: P) -> Result<PathFile> {
         symlink_file(&self, &dst).map_err(|err| {
             Error::new(
                 err,
                 &format!("linking to {} from", dst.as_ref().display()),
```

Comment:
(while we're here, this message is confusing. Shouldn't it be `linking from {} to`?